consensus/src/threshold_simplex/actors/voter/actor.rs

```diff
@@ -64,37 +60,39 @@ struct Round<
 > {
     start: SystemTime,
     context: E,
+    reporter: R,
     supervisor: S,
 
-    leader: Option<C::PublicKey>,
-
     view: View,
+    participants: usize,
+
+    leader: Option<C::PublicKey>,
     leader_deadline: Option<SystemTime>,
     advance_deadline: Option<SystemTime>,
     nullify_retry: Option<SystemTime>,
 
     // Track one proposal per view (only matters prior to notarization)
-    proposal: Option<(Sha256Digest /* proposal */, Parsed<wire::Proposal, D>)>,
+    proposal: Option<Proposal<D>>,
     requested_proposal: bool,
     verified_proposal: bool,
 
     // Track notarizes for all proposals (ensuring any participant only has one recorded notarize)
-    notaries: HashMap<u32, Sha256Digest>,
-    notarizes: HashMap<Sha256Digest, HashMap<u32, Parsed<wire::Notarize, D>>>,
-    notarization: Option<Parsed<wire::Notarization, D>>,
+    notarized_proposals: HashMap<Proposal<D>, Vec<u32>>,
+    notarizes: Arc<Vec<Option<Notarize<D>>>>,
+    notarization: Option<Notarization<D>>,
     broadcast_notarize: bool,
     broadcast_notarization: bool,
 
     // Track nullifies (ensuring any participant only has one recorded nullify)
-    nullifies: HashMap<u32, wire::Nullify>,
-    nullification: Option<wire::Nullification>,
+    nullifies: Arc<HashMap<u32, Nullify>>,
```

Comment:
Haven't read through the file yet but curious why we started using `Arc` here. AFAIK we basically don't have to use it anywhere else